dev-docs/telemetry-selectors.md

```diff
@@ -0,0 +1,112 @@
+# Telemetry selectors
+
+The router has now plenty of [selectors](https://www.apollographql.com/docs/router/configuration/telemetry/instrumentation/selectors/) which are really helpful for our users to customize the telemetry as they want. For example with selectors they're able to add custom attributes on [spans](https://www.apollographql.com/docs/router/configuration/telemetry/instrumentation/spans), or create custom [instruments](https://www.apollographql.com/docs/router/configuration/telemetry/instrumentation/instruments) to have custom metrics but also conditionnaly [log events](https://www.apollographql.com/docs/router/configuration/telemetry/instrumentation/events/). For example, existing selectors are `request_header` to fetch the value of a specific request header, `trace_id` to get the current trace id, `supergraph_query` to get the current query executed at the supergraph level.
+
+## Goal
+
+Both selectors can be used as a value for an attribute or a metric value but also as a condition. What we would like to achieve in the future is to provide a great flexibility to our users to not have to create our own metrics when adding new features. For example let's talk about authentication features, we could provide a selector called `authenticated` which would return a boolean to know if the user executing the query is authenticated or not, thanks to this selector we would end up for our users to create their own metrics to know for example how many authenticated metrics are executed. If you combine this with a selector giving the operation cost we could create an histogram indicating what's the distribution of operation cost for authenticated queries.
```

Comment:
```suggestion
Selectors may be used as a value for an attribute or a metric value and also as a condition. What we would like to achieve in the future is to provide enough flexibility to users so that we can reduce the number of metrics provided by default in the router when adding new features. For example, let's talk about authentication features. We could provide a selector called `authenticated` which would return a boolean specifying that the user executing the query is authenticated (or not). This selector would provide the ability for our users to create their own metrics which specified, for example, how many authenticated metrics are executed. If you combine this with a selector giving the operation cost we could create an histogram indicating the distribution of operation cost for authenticated queries.
```